lib/tjbot.js

```diff
@@ -92,85 +84,78 @@ function TJBot(hardware, configuration, credentials) {
-        var creds = credentials['conversation'];
-        this._createServiceAPI('conversation', creds);
-    }
+    if (credentials != undefined) {
+        // > conversation
+        if (credentials.hasOwnProperty('conversation') && credentials.conversation.password != '') {
+            var creds = credentials['conversation'];
+            this._createServiceAPI('conversation', creds);
+        }
 
-    // > speech to text
-    if (credentials != undefined && credentials.hasOwnProperty('speech_to_text')) {
-        var creds = credentials['speech_to_text'];
-        this._createServiceAPI('speech_to_text', creds);
-    }
+        // > speech to text
+        if (credentials.hasOwnProperty('speech_to_text') && credentials.speech_to_text.password != '') {
+            var creds = credentials['speech_to_text'];
+            this._createServiceAPI('speech_to_text', creds);
+        }
 
-    // > text to speech
-    if (credentials != undefined && credentials.hasOwnProperty('text_to_speech')) {
-        var creds = credentials['text_to_speech'];
-        this._createServiceAPI('text_to_speech', creds);
-    }
+        // > text to speech
+        if (credentials.hasOwnProperty('text_to_speech') && credentials.text_to_speech.password != '') {
+            var creds = credentials['text_to_speech'];
+            this._createServiceAPI('text_to_speech', creds);
+        }
 
-    // > tone analyzer
-    if (credentials != undefined && credentials.hasOwnProperty('tone_analyzer')) {
-        var creds = credentials['tone_analyzer'];
-        this._createServiceAPI('tone_analyzer', creds);
-    }
+        // > language translator
+        if (credentials.hasOwnProperty('language_tranlator') && credentials.language_tranlator.password != '') {
+            var creds = credentials['language_tranlator'];
+            this._createServiceAPI('language_tranlator', creds);
+        }
 
-    // > visual recognition
-    if (credentials != undefined && credentials.hasOwnProperty('visual_recognition')) {
-        var creds = credentials['visual_recognition'];
-        this._createServiceAPI('visual_recognition', creds);
+        // > tone analyzer
+        if (credentials.hasOwnProperty('tone_analyzer') && credentials.tone_analyzer.password != '') {
+            var creds = credentials['tone_analyzer'];
+            this._createServiceAPI('tone_analyzer', creds);
+        }
+
+        // > visual recognition
+        if (credentials.hasOwnProperty('visual_recognition') && credentials.visual_recognition.key != '') {
+            var creds = credentials['visual_recognition'];
+            this._createServiceAPI('visual_recognition', creds);
+        }
     }
 
-    if (this.configuration.verboseLogging) {
+    if (this.config.verboseLogging) {
         console.log("Hello from TJBot!");
     }
 }
 
 /**
  * TJBot module version
  */
-TJBot.prototype.version = 'v1';
+TJBot.prototype.version = 'v1.0.1';
```

Comment:
Just some comments: the right place to add nodejs version is on package.json, attribute version, following this format:
"version": "1.0.1"

Example of Watson API:
https://github.com/watson-developer-cloud/node-sdk/blob/master/package.json

I don't know why "TJBot.prototype.version" is being used. So to avoid any problem I just keep it.
If you agree to change, I could change it.